src/main/java/gov/usgs/wma/waterdata/collections/CollectionsController.java

```diff
@@ -100,19 +104,18 @@ public String getOgcCollectionFeatures(
 			@RequestParam(value = "bbox", required = false) List<String> bbox,
 			@PathVariable(value = "collectionId") String collectionId, HttpServletResponse response) {
 
-		int count = collectionsDao.getCollectionFeatureCount(collectionsParams.buildParams(collectionId));
 
 		String rtn = null;
-		if (limit == 0 || startIndex >= count) {
-			response.setStatus(HttpStatus.NOT_FOUND.value());
-		} else {
-			rtn = collectionsDao.getCollectionFeaturesJson(
-					collectionsParams.buildParams(collectionId, limit, startIndex, bbox, count));
-			if (rtn == null) {
-				response.setStatus(HttpStatus.NOT_FOUND.value());
-			}
+		// requesting less than one is a non-starter
+		resultOr404(response, limit>0 ?"Good" :null);
+		if (response.getStatus() == HttpServletResponse.SC_OK) {
```

Comment:
My gut tells me relying on the response status code like this is not good and should be avoided at all costs. At the very least, I think you only have one opportunity to set it.

Reply:
This was my attempt to refactor to DRY code of prior work. We have a condition repeated in the code to check for null response and set the response code to 404 on null. The new method has that decision in one place. When I wrote it, each time it was called only once. When I updated to the new HEAD this seemed to be necessary to use it in the new method. If we can only set it once then we will certainly have to revise it.
_or_
We could just revert it.

Reply:
I have figured out a revision that I am testing. We only need to set that status once. Once it is set we could return from the call.

Reply:
The tests passed.